rows.go

```diff
@@ -146,6 +146,15 @@ func (r *rows) Next(dest []driver.Value) error {
 			} else {
 				dest[i] = nil
 			}
+		case sppb.TypeCode_JSON:
```

Comment:
This and the below addition for `ARRAY<JSON>` are the only functional changes that are needed to support the new data type. All other changes are tests (and workarounds for tests on the emulator that does not yet support JSON).